### PR TITLE
Add KKP 2.22.0 changelog

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**:
This PR adds the 2.22.0 changelog.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubermatic/release/issues/4

**What type of PR is this?**
/kind documentation

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: docs/changelogs/CHANGELOG-2.22.md [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+- Bugfix for KubeVirt infra CSI token creation due to auto-creation disabled in k8s 1.24 ([#10908](https://github.com/kubermatic/kubermatic/pull/10908))
+- Introduction of KubeVirt default instance types and instance preferences that will replace virtual machine presets ([#11025](https://github.com/kubermatic/kubermatic/pull/11025))
+- Fix wrong quota filtering when `VirtualMachineInstancePreset.spec.cpu` has no quantity but only other fields ([#11046](https://github.com/kubermatic/kubermatic/pull/11046))
+- Add support for ToplogySpreadConstraint for Kubevirt VM ([#11114](https://github.com/kubermatic/kubermatic/pull/11114))
+- Add support for instancetype/preference ([#11182](https://github.com/kubermatic/kubermatic/pull/11182))
+- Update KubeVirt CCM to v0.4.0 and kubevirt.io/api to v0.58.0 ([#11249](https://github.com/kubermatic/kubermatic/pull/11249))
+- Add Images option in KubeVirt DataCenter which supports download and caching of VM images from HTTP source ([#11302](https://github.com/kubermatic/kubermatic/pull/11302))
+- Add DataVolumeSize field to seed and ImageCloningEnabled to cluster ([#11651](https://github.com/kubermatic/kubermatic/pull/11651))
```
Comment (mfranczy) [inline docs/changelogs/CHANGELOG-2.22.md]: Please remove this as the custom images were not delivered.

Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md]: Removed both lines, thanks for the heads up.

**Comment by ahmedwaleedmalik** [comment]
@embik we need a note regarding machine-controller user-data deprecation and possible removal in the next(or the one after it) release.

**Comment by embik** [comment]
> @embik we need a note regarding machine-controller user-data deprecation and possible removal in the next(or the one after it) release.
> 
> We don't have this as a release note in a PR so maybe just adding it directly in the CHANGELOG would be enough?!

Sure thing, let me add something.

**Comment by ahmedwaleedmalik** [comment]
I think we need to highlight the fact that docker will no longer be supported with KKP 2.22 and users need to migrate their clusters AND cluster templates before the migration.

**Comment by embik** [comment]
> I think we need to highlight the fact that docker will no longer be supported with KKP 2.22 and users need to migrate their clusters AND cluster templates before the migration.

IMO this would be something for https://github.com/kubermatic/docs/issues/1327, not necessarily the changelog.

**Comment by ahmedwaleedmalik** [comment]
> > I think we need to highlight the fact that docker will no longer be supported with KKP 2.22 and users need to migrate their clusters AND cluster templates before the migration.
> 
> IMO this would be something for [kubermatic/docs#1327](https://github.com/kubermatic/docs/issues/1327), not necessarily the changelog.

I would expect it to be somewhere in the changelogs at least. But yeah the migration part can be covered with docs.

File: docs/changelogs/CHANGELOG-2.22.md [hunk 2]
```diff
@@ -0,0 +1,481 @@
+# Kubermatic 2.22
+
+- [v2.22.0](#v2220)
+
+## [v2.22.0](https://github.com/kubermatic/kubermatic/releases/tag/v2.22.0)
+
+Before upgrading, make sure to read the [general upgrade guidelines](https://docs.kubermatic.com/kubermatic/v2.22/tutorials-howtos/upgrading/). Consider tweaking `seedControllerManager.maximumParallelReconciles` to ensure usercluster reconciliations will not cause resource exhaustion on seed clusters.
```
Comment (pkprzekwas) [inline docs/changelogs/CHANGELOG-2.22.md]: Not sure if it should be `usercluster`, `user cluster`, `user cluster's` or `user-cluster`...
```suggestion
Before upgrading, make sure to read the [general upgrade guidelines](https://docs.kubermatic.com/kubermatic/v2.22/tutorials-howtos/upgrading/). Consider tweaking `seedControllerManager.maximumParallelReconciles` to ensure user cluster reconciliations will not cause resource exhaustion on seed clusters.
```

Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md]: "user cluster" is the best out of them I think.

File: docs/changelogs/CHANGELOG-2.22.md [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+    - OSP and OSC resources have been moved to user clusters. KKP will take care of migrating the existing resources
+    - Custom OperatingSystemProfiles should now be created for the kind `CustomOperatingSystemProfile` instead of `OperatingSystemProfile` in the seed namespace
+- Action required on applications: helm has been bumped to version 3.11. Due to a [CVE](https://github.com/helm/helm/security/advisories/GHSA-pwcw-6f5g-gxf8) in the previous version, the helm template function [getHostByName](https://helm.sh/docs/chart_template_guide/function_list/#network-functions has been disabled (function will return an empty string).  If you want to enable this function, you have to set `deployOptions.helm.enableDNS` to true and *verify* this function 'getHostByName' is not being used in a chart to disclose any information you do not want to be passed to DNS servers.(c.f. CVE-2023-25165) ([#11887](https://github.com/kubermatic/kubermatic/pull/11887))
+
+### API Changes
```
Comment (pkprzekwas) [inline docs/changelogs/CHANGELOG-2.22.md:141]: I know that CRDs are technically our API, but maybe we should rename it to `CRD Changes`?

Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md:141]: Let's keep it as "API Changes" - e.g. the initial machinedeployment annotation is technically not part of a CRD, so I'd like to keep this section flexible.

**Comment by embik** [comment]
/hold

Just making sure to merge this timed correctly.

File: docs/changelogs/CHANGELOG-2.22.md [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+- Add separate list and details pages for KubeOne clusters ([#5412](https://github.com/kubermatic/dashboard/pull/5412))
+- Add wizard to import KubeOne clusters ([#5362](https://github.com/kubermatic/dashboard/pull/5362))
+- Add support to import AWS KubeOne clusters ([#5362](https://github.com/kubermatic/dashboard/pull/5362))
+- Add support to import Azure KubeOne clusters ([#5488](https://github.com/kubermatic/dashboard/pull/5488))
+- Add support to import GCP KubeOne clusters ([#5460](https://github.com/kubermatic/dashboard/pull/5460))
+- Add support to migrate KubeOne cluster container runtime ([#5499](https://github.com/kubermatic/dashboard/pull/5499))
+- Add support to select preset in credentials step of KubeOne wizard ([#5504](https://github.com/kubermatic/dashboard/pull/5504))
+- Add support to upgrade KubeOne machine deployment version ([#5561](https://github.com/kubermatic/dashboard/pull/5561))
+
+#### Web Terminal
```
Comment (csengerszabo) [inline docs/changelogs/CHANGELOG-2.22.md:65]: Only one PR for web terminal?

File: docs/changelogs/CHANGELOG-2.22.md [hunk 5]
```diff
@@ -0,0 +1,481 @@
...
+#### Presets
+
+- Add `.spec.projects` field to `Preset` resources to allow binding Presets to specific projects ([#11100](https://github.com/kubermatic/kubermatic/pull/11100))
+
+#### OIDC
+
+- Add groups in OIDC kubeconfig ([#11121](https://github.com/kubermatic/kubermatic/pull/11121))
+- Add `OIDCProviderConfiguration` to Seed's spec allowing to configure dedicated OIDC provider for each Seed ([#11668](https://github.com/kubermatic/kubermatic/pull/11668))
+
+#### Resource Quotas (EE)
```
Comment (csengerszabo) [inline docs/changelogs/CHANGELOG-2.22.md:114]: We should also add the dyanmic quota feedback on the UI while creating a cluster

File: docs/changelogs/CHANGELOG-2.22.md [hunk 6]
```diff
@@ -0,0 +1,481 @@
...
+
+- 1.24.8
+- 1.24.9
+- 1.24.10
+- 1.25.4
+- 1.25.6
+- 1.25.6
+- 1.26.1
+
+### Highlights
```
Comment (csengerszabo) [inline docs/changelogs/CHANGELOG-2.22.md:31]: Let's also add the new Admin panel and Project panel sidebars (the groupings and toggle feature makes the feature list more structured)

File: docs/changelogs/CHANGELOG-2.22.md [hunk 7]
```diff
@@ -0,0 +1,481 @@
...
+#### OIDC
+
+- Add groups in OIDC kubeconfig ([#11121](https://github.com/kubermatic/kubermatic/pull/11121))
+- Add `OIDCProviderConfiguration` to Seed's spec allowing to configure dedicated OIDC provider for each Seed ([#11668](https://github.com/kubermatic/kubermatic/pull/11668))
+
+#### Resource Quotas (EE)
+
+- Add a default project resource quota setting which can be set in KKP's global `KubermaticSettings`. By managing the default quota, for all the projects which do not have a custom quota already set, their ResourceQuota is created/updated/deleted ([#11582](https://github.com/kubermatic/kubermatic/pull/11582))
+
+### Breaking Changes
```
Comment (csengerszabo) [inline docs/changelogs/CHANGELOG-2.22.md:121]: I think we should also add these here:
- Removing Docker
- Deprecation of PodSecurityPolicy

File: docs/changelogs/CHANGELOG-2.22.md (hunk 6 above)
Comment (csengerszabo) [reply docs/changelogs/CHANGELOG-2.22.md:31]: - Including MLA within KKP installer could also be highlighted here as well
- Cilium CNI customization and management on Applications base could also be

File: docs/changelogs/CHANGELOG-2.22.md (hunk 7 above)
Comment (csengerszabo) [reply docs/changelogs/CHANGELOG-2.22.md:121]: Removing the API and adding it to the Private Dashboard repo should also be mentioned as a breaking change as well I think

**Review by csengerszabo — CHANGES_REQUESTED** [review]

File: docs/changelogs/CHANGELOG-2.22.md [hunk 8]
```diff
@@ -0,0 +1,481 @@
...
+    - [CVE-2022-3162: Unauthorized read of Custom Resources](https://groups.google.com/g/kubernetes-announce/c/oR2PUBiODNA/m/tShPgvpUDQAJ)
+    - [CVE-2022-3294: Node address isn't always verified when proxying](https://groups.google.com/g/kubernetes-announce/c/eR0ghAXy2H8/m/sCuQQZlVDQAJ)
+- Add support for Kubernetes v1.24.9 ([#11553](https://github.com/kubermatic/kubermatic/pull/11553))
+- Add support for Kubernetes 1.25 ([#11049](https://github.com/kubermatic/kubermatic/pull/11049))
+- Add support for Kubernetes 1.26 ([#11621](https://github.com/kubermatic/kubermatic/pull/11621))
+- Support for Kubernetes 1.22 and 1.23 user clusters has been removed; User clusters remaining on 1.23 will be automatically upgraded with this KKP version ([#11286](https://github.com/kubermatic/kubermatic/pull/11286), [#11767](https://github.com/kubermatic/kubermatic/pull/11767))
+- Allow Kubernetes version upgrade for clusters with non-amd64 nodes & Canal CNI and IPVS for all Kubernetes versions ([#11765](https://github.com/kubermatic/kubermatic/pull/11765))
+- Add support for Kubernetes v1.24.10, v1.25.6, and v1.26.1 ([#11859](https://github.com/kubermatic/kubermatic/pull/11859))
```
Comment (xrstf) [inline docs/changelogs/CHANGELOG-2.22.md]: Can the last item on this list simply be merged into the others? We have 3 items that each mention support for a new 1.24.x release, usually we only mentioned the most recent support patch release and do not tell users "please update to this one, even though this other one is newer and also supported" ^^

File: docs/changelogs/CHANGELOG-2.22.md [hunk 9]
```diff
@@ -0,0 +1,481 @@
...
+- Update seed and cluster to config StorageClasses from seed ([#11701](https://github.com/kubermatic/kubermatic/pull/11701))
+- Graceful user cluster workload eviction in case of bare-metal infra node draining ([#11588](https://github.com/kubermatic/kubermatic/pull/11588))
+- Split CSI driver deployment between user and infra cluster and remove user cluster CSI access to KubeVirt cluster ([#11370](https://github.com/kubermatic/kubermatic/pull/11370))
+- Fix KubeVirt LB issue (wrong custer-isolation netpol): LB not accessible from outside user-cluster ([#11640](https://github.com/kubermatic/kubermatic/pull/11640))
+- Change CustomNetworkPolicies type (extract name) ([#11666](https://github.com/kubermatic/kubermatic/pull/11666))
+- Fix missing ccmClusterName feature in KubeVirt clusters upgraded to KKP 2.22 ([#11844](https://github.com/kubermatic/kubermatic/pull/11844))
+
+#### KubeOne Cluster Support
+
+- Update KubeOne image to v1.5.3 for KubeOne external cluster controller ([#11388](https://github.com/kubermatic/kubermatic/pull/11388))
```
Comment (xrstf) [inline docs/changelogs/CHANGELOG-2.22.md]: Still true after https://github.com/kubermatic/kubermatic/pull/11914 ?

File: docs/changelogs/CHANGELOG-2.22.md [hunk 10]
```diff
@@ -0,0 +1,481 @@
...
+- Add support to import AWS KubeOne clusters ([#5362](https://github.com/kubermatic/dashboard/pull/5362))
+- Add support to import Azure KubeOne clusters ([#5488](https://github.com/kubermatic/dashboard/pull/5488))
+- Add support to import GCP KubeOne clusters ([#5460](https://github.com/kubermatic/dashboard/pull/5460))
+- Add support to migrate KubeOne cluster container runtime ([#5499](https://github.com/kubermatic/dashboard/pull/5499))
+- Add support to select preset in credentials step of KubeOne wizard ([#5504](https://github.com/kubermatic/dashboard/pull/5504))
+- Add support to upgrade KubeOne machine deployment version ([#5561](https://github.com/kubermatic/dashboard/pull/5561))
+
+#### Web Terminal
+
+- Add Web terminal support for user clusters ([#4492](https://github.com/kubermatic/dashboard/pull/4492))
```
Comment (xrstf) [inline docs/changelogs/CHANGELOG-2.22.md]: `..... This allows users to connect to their clusters from the KKP dashboard, using only their browser.`

File: docs/changelogs/CHANGELOG-2.22.md [hunk 11]
```diff
@@ -0,0 +1,481 @@
...
+    - OSP and OSC resources have been moved to user clusters. KKP will take care of migrating the existing resources
+    - Custom OperatingSystemProfiles should now be created for the kind `CustomOperatingSystemProfile` instead of `OperatingSystemProfile` in the seed namespace
+- Action required on applications: helm has been bumped to version 3.11. Due to a [CVE](https://github.com/helm/helm/security/advisories/GHSA-pwcw-6f5g-gxf8) in the previous version, the helm template function [getHostByName](https://helm.sh/docs/chart_template_guide/function_list/#network-functions has been disabled (function will return an empty string).  If you want to enable this function, you have to set `deployOptions.helm.enableDNS` to true and *verify* this function 'getHostByName' is not being used in a chart to disclose any information you do not want to be passed to DNS servers.(c.f. CVE-2023-25165) ([#11887](https://github.com/kubermatic/kubermatic/pull/11887))
```
Comment (xrstf) [inline docs/changelogs/CHANGELOG-2.22.md]: Link is broken. Also there's a double space in this list item.

File: docs/changelogs/CHANGELOG-2.22.md [hunk 12]
```diff
@@ -0,0 +1,481 @@
...
+
+- Update metering to version 1.0.1 ([#11282](https://github.com/kubermatic/kubermatic/pull/11282))
+    - Add average-used-cpu-millicores to Cluster and Namespace reports
+    - Add average-available-cpu-millicores add average-cluster-machines field to Cluster reports
+    - Fix a bug that causes wrong values if metric is not continuously present for the aggregation window 
+
+### Miscellaneous
+
+- Update codebase to Kubernetes 1.26 ([#11600](https://github.com/kubermatic/kubermatic/pull/11600))
+- Update Go to version 1.19.4 ([#11530](https://github.com/kubermatic/kubermatic/pull/11530))
```
Comment (xrstf) [inline docs/changelogs/CHANGELOG-2.22.md]: Soon to be changed.

File: docs/changelogs/CHANGELOG-2.22.md (hunk 9 above)
Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md]: No, but the final version to be included isnt merged yet.

File: docs/changelogs/CHANGELOG-2.22.md (hunk 8 above)
Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md]: fair point

File: docs/changelogs/CHANGELOG-2.22.md (hunk 4 above)
Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md:65]: only one with a changelog entry.

File: docs/changelogs/CHANGELOG-2.22.md (hunk 7 above)
Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md:121]: there is also Removals later down the document. For API I couldn't get a definitive answer on sig-release.

File: docs/changelogs/CHANGELOG-2.22.md (hunk 12 above)
Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md]: true, needs to be updated.

File: docs/changelogs/CHANGELOG-2.22.md (hunk 11 above)
Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md]: Fixed, thanks.

File: docs/changelogs/CHANGELOG-2.22.md (hunk 12 above)
Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md]: Updated.

File: docs/changelogs/CHANGELOG-2.22.md (hunk 10 above)
Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md]: Added, thanks!

File: docs/changelogs/CHANGELOG-2.22.md (hunk 8 above)
Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md]: Updated this section.

File: docs/changelogs/CHANGELOG-2.22.md (hunk 6 above)
Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md:31]: I've added CNI management via Applications. While MLA integration in the installer is nice, it doesn't change much for day-to-day usage and we're highlighting this change on the upgrade guide already.

File: docs/changelogs/CHANGELOG-2.22.md (hunk 5 above)
Comment (embik) [reply docs/changelogs/CHANGELOG-2.22.md:114]: Done.

**Review by csengerszabo — COMMENTED** [review]
/lgtm

**Review by csengerszabo — APPROVED** [review]
/lgtm

**Pull request merged**